src/checkbox.jsx

```diff
@@ -55,7 +55,12 @@ const Checkbox = React.createClass({
   //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: newMuiTheme,
+      switched: typeof nextProps.checked === 'boolean' && this.props.checked !== nextProps.checked
+        ? nextProps.checked
+        : this.state.switched,
+    });
```

Comment:
I'm not sure about the typeof

https://github.com/facebook/react-native/blob/master/Examples/UIExplorer/CameraRollView.ios.js#L126
https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/ListView/ListView.js#L280